### DISCUSSION TITLE
How to render a template using Node?

### DISCUSSION THREAD

**Discussion by giopetris** [resend/react-email#569] [Q&A]
How can I render a template through Node and also passing props to the template?

**Comment by supagu** [comment]
I also am trying to do this. I have AWS Lambda using node.js and so getting a jsx error following the AWS SES example.
Is it just a matter of using react without jsx code?

**Comment by zachgoll** [comment]
Hey @giopetris 👋 

Should be able to do it like this:

```ts
import EmailComponent from "path/to/component";
import { render } from "@react-email/render";

const html = render(EmailComponent({prop1: 'value'}))
```

**Reply by bastiaanv** [reply]
You would also need to install babel. More info see this link: https://react.email/docs/integrations/overview

**Comment by Iamstanlee** [comment]
> Hey @giopetris 👋
> 
> Should be able to do it like this:
> 
> ```ts
> import EmailComponent from "path/to/component";
> import { render } from "@react-email/render";
> 
> const html = render(EmailComponent({prop1: 'value'}))
> ```

Hey @zachgoll 👋 , this didnt work for me using using ts + nextjs, got the error

`error - node_modules/@react-email/render/dist/index.mjs (10:17) @ render
error - react_dom_server__WEBPACK_IMPORTED_MODULE_0__.renderToStaticMarkup is not a function`

Can you advice? thanks

**Reply by bukinoshita** [reply]
Here we have an example using React Email + Next.js

https://github.com/resendlabs/resend-examples/tree/main/with-react-email

**Comment by supagu** [comment]
I'm not sure you can use EmailComponent directly. You need to use:

React.createElement(EmailComponent, ....

I'm using a library called htm which replaces JSX and saves the verbosity of React.createElement. Here is my sample:

`const React = require('react')
const { html } = require('htm/react')
const { Html } = require('@react-email/html')
const { Button } = require('@react-email/button')
const { Link } = require('@react-email/link')
const { Text } = require('@react-email/text')

const Email = ({ url }) => {
  return (html`
    <${Html}>
      <${Text}>
        You have been invited to connect your account to qwe
      <//>
      <${Button} href={${url}} style={{ color: '#61dafb' }}>
        Click me
      <//>
      <${Link} href="https://example.com">Click here to login<//>
    <//>
  `)
}

const emailHtml = render(html`<${Email} url="https://example.com" />`)`

**Comment by bukinoshita** [comment]
We have an example with Next.js + React Email + Resend [here](https://github.com/resendlabs/resend-examples/tree/main/with-react-email). Let me know if that helps you.

**Reply by jinsley8** [reply]
@bukinoshita Can you add a README to that example with details about how this runs with Next.js?

Should `/transactional` folder be committed to git and deployed to Vercel along with the app?

**Comment by clemenscodes** [comment]
Using `React.createElement`

```tsx
import { Text } from '@react-email/text';

export type EmailProps = { content: string };

export const Email: React.FC<EmailProps> = ({ content }) => {
    return <Text>{content}</Text>;
};
```
Now in a next.js API route:

```ts
const props = { content: "Prop passed to component in Node.js" };
const emailElement = React.createElement(Email, props);
const html = render(emailElement, {});
```

Now html has the value `<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><p data-id="react-email-text" style="font-size:14px;line-height:24px;margin:16px 0">Prop passed to component in Node.js</p>`
